File: client-android/app/src/main/java/app/serenada/android/call/CallUiState.kt
package app.serenada.android.call

data class CallUiState(
    val phase: CallPhase = CallPhase.Idle,
    val roomId: String? = null,
    val statusMessageResId: Int? = null,
    val errorMessageResId: Int? = null,
    val errorMessageText: String? = null,
    val isHost: Boolean = false,
    val participantCount: Int = 0,
    val localAudioEnabled: Boolean = true,
    val localVideoEnabled: Boolean = true,
    val remoteVideoEnabled: Boolean = false,
    val isReconnecting: Boolean = false,
    val isSignalingConnected: Boolean = false,
    val iceConnectionState: String = "NEW",
    val connectionState: String = "NEW",
    val signalingState: String = "STABLE",
    val activeTransport: String? = null,
    val webrtcStatsSummary: String = "",
    val realtimeCallStats: RealtimeCallStats? = null,
    val isFrontCamera: Boolean = true,
    val isScreenSharing: Boolean = false,
    val localCameraMode: LocalCameraMode = LocalCameraMode.SELFIE,
    val isFlashAvailable: Boolean = false,
    val isFlashEnabled: Boolean = false
)

File: client-ios/Sources/Core/Models/CallUiState.swift
import Foundation

struct CallUiState: Equatable {
    var phase: CallPhase = .idle
    var roomId: String?
    var statusMessage: String?
    var errorMessage: String?
    var isHost: Bool = false
    var participantCount: Int = 0
    var localAudioEnabled: Bool = true
    var localVideoEnabled: Bool = true
    var remoteVideoEnabled: Bool = false
    var isReconnecting: Bool = false
    var isSignalingConnected: Bool = false
    var iceConnectionState: String = "NEW"
    var connectionState: String = "NEW"
    var signalingState: String = "STABLE"
    var activeTransport: String?
    var webrtcStatsSummary: String = ""
    var realtimeStats: RealtimeCallStats = .empty
    var isFrontCamera: Bool = true
    var isScreenSharing: Bool = false
    var localCameraMode: LocalCameraMode = .selfie
    var cameraZoomFactor: Double = 1
    var isFlashAvailable: Bool = false
    var isFlashEnabled: Bool = false
}
